packages/react/src/hooks/contracts/usePrepareContractWrite.ts

```diff
@@ -91,18 +98,21 @@ export function usePrepareContractWrite({
   onSettled,
   onSuccess,
 }: UsePrepareContractWriteArgs & UsePrepareContractWriteConfig) {
-  const chainId = useChainId()
-  const { data: signer } = useSigner<providers.JsonRpcSigner>()
+  const activeChainId = useChainId()
+  const { data: signer } = useSigner<providers.JsonRpcSigner>({
+    chainId: chainId ?? activeChainId,
+  })
 
   const prepareContractWriteQuery = useQuery(
     queryKey(
       {
         addressOrName,
         functionName,
+        chainId,
         args,
         overrides,
       },
-      { chainId, signerAddress: signer?._address },
+      { activeChainId, signerAddress: signer?._address },
```

Comment:
What's the difference between `chainId` and `activeChainId`? Could they collapse into one with `const activeChainId = useChainId({ chainId })`?

Reply:
We need the `activeChainId` to cache bust.